src/pages/Main/MainSlide/MainSlide.js

```diff
@@ -1,22 +1,109 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import styled from 'styled-components';
 
 import MainSlideItem from './MainSlideItem/MainSlideItem';
+import MainSlideDot from './MainSlideDot/MainSlideDot';
+
+import { SLIDE_BG_COLOR } from './slideBgColor';
+
+let isMoving = true;
+
+function MainSlide({ slide }) {
+  const [slideSpot, setSlideSpot] = useState(-100);
+  const [isTransitionOn, setIsTransitionOn] = useState(true);
+
+  const initialMount = useRef(true);
+  const timeoutRef = useRef(null);
+
+  const handleSlideAutoMove = () => {
+    setIsTransitionOn(true);
+    setSlideSpot(prev => prev - 100);
+  };
+
+  const resetTimeout = () => {
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
+    }
+  };
+
+  useEffect(() => {
+    resetTimeout();
+    timeoutRef.current = setTimeout(() => handleSlideAutoMove(), 3000);
+
+    return () => {
+      resetTimeout();
+    };
+  }, [slideSpot]);
+
+  const handleSlideBtn = direction => {
+    if (!isMoving) return;
+    isMoving = true;
+
+    setIsTransitionOn(true);
+
+    if (direction === 'left') {
+      setSlideSpot(prev => prev - 100);
+    }
+
+    if (direction === 'right') {
+      setSlideSpot(prev => prev + 100);
+    }
+
+    isMoving = false;
+    setTimeout(() => {
+      isMoving = true;
+    }, 1000);
+  };
+
+  const handleSlideEnd = () => {
+    if (slideSpot === 0) {
+      setIsTransitionOn(false);
+      setSlideSpot((slide.length - 2) * -100);
+    } else if (slideSpot === (slide.length - 1) * -100) {
+      setIsTransitionOn(false);
+      setSlideSpot(-100);
+    }
+  };
+
+  useEffect(() => {
+    if (initialMount.current) {
+      initialMount.current = false;
+    } else {
+      setTimeout(handleSlideEnd, 1001);
+    }
+  }, [slideSpot]);
 
-function MainSlide() {
   return (
     <SlideBox>
-      <ArrowBtn className="left">
-        <i className="fas fa-chevron-left"></i>
-      </ArrowBtn>
-      <SlideInner>
-        {[...Array(7)].map((item, i) => (
-          <MainSlideItem key={i} idx={i} item={item} />
+      <ArrowBtn left onClick={() => handleSlideBtn('left')}></ArrowBtn>
+      <SlideInner
+        isTransitionOn={isTransitionOn}
+        quantity={slide.length}
+        slideSpot={slideSpot}
+      >
+        {slide.map((item, i) => (
+          <MainSlideItem
+            data={item}
+            key={i}
+            bgIdx={
+              (i === 0 && SLIDE_BG_COLOR.length - 1) ||
+              (i === slide.length - 1 && '0') ||
+              i - 1
+            }
+          />
         ))}
       </SlideInner>
-      <ArrowBtn className="right">
-        <i className="fas fa-chevron-right"></i>
-      </ArrowBtn>
+      <ArrowBtn right onClick={() => handleSlideBtn('right')}></ArrowBtn>
+      <DotBox>
+        {[...Array(slide.length - 2)].map((_dot, i) => (
```

Comment:
```suggestion
        {[...Array(slide.length - 2)].map((_, i) => (
```
안쓰는 매개변수는 그냥 _로 처리하는것도 좋아보입니다!